pkg/agent/watch.go

```diff
@@ -202,6 +204,22 @@ func (e vmEvent) Format(state fmt.State, verb rune) {
 	}
 }
 
+// extractAutoscalingBounds extracts the ScalingBounds from a VM's autoscaling
+// annotation. We have defined this function for the purpose of metrics, so we
+// could track scaling bounds changes on a VM. api.ExtractVmInfo doesn't provide
+// this data, since its CPU and memory values might come from vm.Spec.
```

Comment:
```suggestion
// extractAutoscalingBounds extracts the ScalingBounds from a VM's autoscaling
// annotation, for the purpose of exposing it in per-VM metrics.
//
// We're not reusing api.ExtractVmInfo even though it also looks at the bounds
// annotation, because its data is less precise - CPU and memory values might
// come from the VM spec without us knowing.
```

I'd reword + simplify + add a line break.

I found it hard to parse the original because logically distinct thoughts started mid-line, and felt there wasn't a clear connection of why we might want to use `api.ExtractVmInfo` in the first place.
IIUC, the bit about "so we could track scaling bound changes on a VM" isn't _quite_ accurate, because in theory that data is fully available in the control plane database - it's just easier to provide it all in one place.

Feel free to reword to fit your style — figured it'd be easier to show what I meant, rather than tell :)